xcm/xcm-builder/src/origin_aliases.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Implementation for ContainsPair<MultiLocation, MultiLocation>.
+
+use frame_support::traits::{Contains, ContainsPair};
+use sp_std::marker::PhantomData;
+use xcm::latest::prelude::*;
+
+/// Alias a Foreign AccountId32 with a local AccountId32 if the Foreign AccountId32 matches the `Prefix` pattern.  
+pub struct AliasForeignAccountId32<Prefix>(PhantomData<Prefix>);
```

Comment:
```suggestion
///
/// Requires that the wrapped origin AccountId32 matches the target AccountId32. 
pub struct AliasForeignAccountId32<Prefix>(PhantomData<Prefix>);
```

Something like this to make this more concrete?